src/main/java/app/exceptions/BaseException.java

```diff
@@ -0,0 +1,16 @@
+package app.exceptions;
+
+/**
+ * This class is the base class for all custom exceptions
+ * related to Duke.
+ */
+public class BaseException extends Exception {
```

Comment:
I like how you had a base exception, and have subsequent exceptions with semantic names which describes what they represent. In addition, headers are informative as they mention situations where such exceptions are used.